### DISCUSSION TITLE
503 HTTP Code - `Retry-After` header

### DISCUSSION THREAD

**Discussion by uwinkler** [TanStack/query#7845] [Q&A]
Hi, 

I was wondering if useQuery is handling 503 HTTP errors. In particular the `retry` option.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503

Common causes are a server that is down for maintenance or that is overloaded. This response should be used for temporary conditions and the [Retry-After](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) HTTP header should, if possible, contain the estimated time for the recovery of the service.

Is tanstack query handling that header as intended?

**Comment by TkDodo** [comment]
TanStack Query doesn't make requests - it manages promises. So no, it doesn't automatically handle HTTP status codes because you can write a queryFn that is simply: `queryFn: () => Promise.resolve(23)`. It's not tied to anything HTTP, it's not tied to graphQL - it's only tied to Promises.

So if you want to handle http headers, you need to extract the headers from the response (depending on how YOU do http requests, because TanStack Query doesn't make the request), return it from `queryFn` alongside your data or as part of your error (more likely with 503) and then implement the `retry` function to act according to that information.

**Reply by uwinkler** [reply]
Thank you! Sorry.. my question was not very specific and a bit miss-leading.  So i guess I could use the `retrayDelay` function such that  the `fetch` should throw an error and that error would than be the `TError` that the `retryDelay` would get, right? 

`retryDelay: number | (retryAttempt: number, error: TError) => number`

**Reply by TkDodo** [reply]
yes
